sys/tracefd/tracefd.c

```diff
@@ -0,0 +1,229 @@
+/* System includes */
+#include <stdarg.h>
+#include <stdlib.h>
+#include <string.h>
+
+/* RIOT includes */
+#include "mutex.h"
+#include "xtimer.h"
+
+#ifndef BOARD_NATIVE
+#  include "mtd_sdcard.h"
+#  include "sdcard_spi.h"
+#  include "sdcard_spi_params.h"
+#  include "fs/fatfs.h"
+#endif
+
+#include "tracefd.h"
+
+#define ENABLE_DEBUG (0)
+#include "debug.h"
+
+#ifndef TRACEFD_ROOT_DIR
+#  ifdef BOARD_NATIVE
+#    define TRACEFD_ROOT_DIR "/tmp"         /**< Base directory of trace files (local directory) */
+#  else /* ! BOARD_NATIVE */
+#    define TRACEFD_ROOT_DIR "/sdcard"      /**< Base directory of trace files (SDCARD mount point) */
+#  endif /* BOARD_NATIVE*/
+#endif
+
+#ifndef TRACEFD_MAX_PATH
+#  define TRACEFD_MAX_PATH 256         /**< Maximum filename length */
+#endif
+
+#ifndef TRACEFD_FLUSH_INTERVAL
+#  define TRACEFD_FLUSH_INTERVAL 1000  /**< Interval in ms between two flush of all files */
+#endif
+
+/**
+ * @brief   Trace file descriptor parameters
+ */
+typedef struct {
+    FILE *fd;                           /**< file descriptor */
+    char filename[TRACEFD_MAX_PATH];    /**< filename */
+    mutex_t lock;                       /**< lock protecting file access */
+} tracefd_context_t;
+
+/* Allocate memory for the tracefd contexts */
+tracefd_context_t tracefd_contexts[TRACEFD_NUMOF];
+
+/* Number of initialized tracefd */
+tracefd_t tracefd_initialized = 0;
+
+/* Files flusher thread PID */
+kernel_pid_t files_flusher_pid = -1;
+
+/* Files flusher thread priority */
+#define FILES_FLUSHER_PRIO (THREAD_PRIORITY_MAIN - 1)
```

Comment:
Dont you think it should be set to the lowest possible priority ?
https://doc.riot-os.org/thread__config_8h.html#a196e3293c94df476dd1dd0007013ad14
And should it be configurable TRACEFD_NUMOF ?

Reply:
Done